homeassistant/components/tankerkoenig/config_flow.py

```diff
@@ -0,0 +1,260 @@
+                        default=user_input.get(
+                            CONF_LOCATION,
+                            {
+                                "latitude": self.hass.config.latitude,
+                                "longitude": self.hass.config.longitude,
+                            },
+                        ),
+                    ): selector({"location": {}}),
+                    vol.Required(
+                        CONF_RADIUS, default=user_input.get(CONF_RADIUS, DEFAULT_RADIUS)
+                    ): selector(
+                        {
+                            "number": {
+                                "min": 0.1,
+                                "max": 25,
+                                "step": 0.1,
+                                CONF_UNIT_OF_MEASUREMENT: LENGTH_KILOMETERS,
+                            }
+                        }
+                    ),
+                }
+            ),
+            errors=errors,
+        )
+
+    def _create_entry(
+        self, data: dict[str, Any], options: dict[str, Any]
+    ) -> FlowResult:
+        return self.async_create_entry(
+            title=data[CONF_NAME],
+            data=data,
+            options=options,
+        )
+
+    async def _get_nearby_stations(self, data: dict[str, Any]) -> dict[str, Any]:
+        """Fetch nearby stations."""
+        try:
+            return await self.hass.async_add_executor_job(
+                getNearbyStations,
+                data[CONF_API_KEY],
+                data[CONF_LOCATION][CONF_LATITUDE],
+                data[CONF_LOCATION][CONF_LONGITUDE],
+                data[CONF_RADIUS],
+                "all",
+                "dist",
+            )
+        except customException as err:
+            return {"ok": False, "message": err, "exception": True}
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle an options flow."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle options flow."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_SCAN_INTERVAL,
+                        default=self.config_entry.options[CONF_SCAN_INTERVAL],
+                    ): selector(
+                        {
+                            "number": {
+                                "min": 5,
+                                "max": 240,
+                                "step": 1,
+                                CONF_UNIT_OF_MEASUREMENT: TIME_MINUTES,
+                            }
+                        }
+                    ),
```

Comment:
We no longer allow for setting a scan interval. Please remove this, users can use the `homeassistant.update_entity` instead.

Reply:
Could you please elaborate a bit more, how (_I guess manual update by_) `homeassistant.update_entity` could replace the automatic data update by a defined interval?

Reply:
`homeassistant.update_entity` can be called to whatever one likes. E.g., using an automation with a custom schedule, that takes into account the moon phase (or whatever magic one wants).

On the config entry options, polling can be disabled, so one can implement their own. This is a generic solution for all.

Reply:
TBH this sounds not really user friendly to me - "hey, disable polling and create some magic on your own ... instead of, set the the polling/update interval in minutes as you need" 🤔 
Nevertheless, is there a generic doc snippet, which I could reference to in the tankerkoenig docs (_like the `integrations/config_flow.md`_) so that the user is aware of this approach?

Reply:
No, as we don't recommend changing the default polling interval. We provide a sane default.

Reply:
@guillempages what do you think, what should be a sane scan interval - 30min?

Reply:
Yes, I think 30 minutes (as with the old yaml based configuration) would be a good default. I assume some users might like to configure more frequent updates (e.g. 15 minutes), but I'd rather have a conservative default.
If someone wants to use a more frequent update, they can set an automation and call the update_entity service there.